seastar/oscar/level1.py

```
@@ -551,59 +570,87 @@ def compute_time_lag_Master_Slave(ds, options):
 
     Parameters
     ----------
-    ds : xarray.Dataset
-        OSCAR SAR dataset
-    options : Time lag computation method. The default is 'from_SAR_time'.
+    ds : ``xarray.Dataset``
+        OSCAR SAR dataset containing orbit, time and baseline information
+    options : ``str``
+        Time lag computation method. The default is 'from_SAR_time'.
 
     Returns
     -------
-    ds : xarray.Dataset
-        OSCAR SAR dataset
-    ds.TimeLag : Time lag tau between Master/Slave images (s)
+
+    TimeLag : ``float``, ``xr.DataArray``
+        Time lag tau between Master/Slave images (s)
 
     """
     if options not in ['from_SAR_time', 'from_aircraft_velocity']:
         raise Exception('Unknown time lag computation method: Please refer to'
                         'function documentation')
     if options == 'from_SAR_time':
-        ds['TimeLag'] = (ds.OrbTimeImage - ds.OrbTimeImageSlave)
+        if 'OrbTimeImageSlave' in ds.data_vars:
+            TimeLag = (ds.OrbTimeImage - ds.OrbTimeImageSlave)
+            TimeLag.attrs['long_name'] = 'Time lag'
+            TimeLag.attrs['description'] =\
+                'Time difference between antenna master and slave.'
+            TimeLag.attrs['units'] = 's'
+        else:
+            TimeLag = xr.DataArray(data=np.NaN)
+            TimeLag.attrs['long_name'] = 'Time lag'
+            TimeLag.attrs['description'] =\
+                'Time difference between antenna master and slave.'\
+                ' Set to NaN as no slave data in beam dataset.'
+            TimeLag.attrs['units'] = 's'
```

is there any meaning to calculate a timelag if there is no Slave?
Should we raise an error instead?

If it is from_aircraft_velocity, we don't have this issue and what you propose looks good to me.